lib/Upcoming.dart

```diff
@@ -359,23 +461,122 @@ class InformationRow extends StatelessWidget {
 }
 
 class RideAction extends StatelessWidget {
-  const RideAction({Key key, this.rideAction, this.color, this.icon})
+  const RideAction({Key key, this.text, this.color, this.icon, this.action})
       : super(key: key);
-
-  final String rideAction;
+  final String text;
   final Color color;
   final IconData icon;
+  final Function action;
 
   @override
   Widget build(BuildContext context) {
-    return Row(
-      mainAxisAlignment: MainAxisAlignment.center,
-      children: <Widget>[
-        Icon(icon, color: color),
-        SizedBox(width: 10),
-        Text(rideAction,
-            style: TextStyle(color: color, fontSize: 18, fontFamily: 'SFPro'))
-      ],
+    return GestureDetector(
+      onTap: () {
+        action();
+      },
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: <Widget>[
+          Icon(icon, color: color),
+          SizedBox(width: 10),
+          Text(text,
+              style: TextStyle(color: color, fontSize: 18, fontFamily: 'SFPro'))
+        ],
+      ),
+    );
+  }
+}
+
+class UpcomingRideCard extends StatelessWidget {
+  UpcomingRideCard(this.ride);
+  final Ride ride;
+
+  final confirmationStyle = TextStyle(
+    fontWeight: FontWeight.w500,
+    fontSize: 10,
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onTap: () {
+        Navigator.push(context,
+            new MaterialPageRoute(builder: (context) => UpcomingRidePage(ride))
+        );
+      },
+      child: Container(
+        constraints: BoxConstraints(maxWidth: MediaQuery.of(context).size.width * 0.65),
+        child: Card(
+          elevation: 3.0,
+          child: Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 16),
+            child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: <Widget>[
+                  ride.type == 'active' ? Text('Ride Confirmed', style: confirmationStyle.copyWith(color: Color(0xFF4CAF50))) :
+                  Text('Ride Requested', style: confirmationStyle.copyWith(color: Color(0xFFFF9800))),
+                  SizedBox(height: 4),
+                  ride.buildStartTime(),
+                  SizedBox(height: 16),
+                  Text('From', style: TextThemes.directionStyle),
+                  Text(ride.startLocation, style: TextThemes.rideInfoStyle),
+                  SizedBox(height: 8),
+                  Text('To', style: TextThemes.directionStyle),
```

Comment:
Good use of a creating your own class to store text styles to avoid repeating code. I have also created one in one of my unpublished PRs, but will just move my text styles into this class as well.

Reply:
Hopefully the merge will be smooth!